AdvancedMessageBoxHelper.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace CnCNet.LauncherStub;
+
+using System.Collections.ObjectModel;
+
+public static class AdvancedMessageBoxHelper
+{
+    public static int? ShowMessageBoxWithSelection(string message, string title, string[] selections)
+    {
+        var msgbox = new AdvancedMessageBox();
+        var model = (AdvancedMessageBoxViewModel)msgbox.DataContext;
+        model.Title = title;
+        model.Message = message;
+
+        var commands = new ObservableCollection<CommandViewModel>();
+        for (int i = 0; i < selections.Length; i++)
+        {
+            int i_copy = i;
+            commands.Add(new CommandViewModel()
+            {
+                Text = selections[i],
+                Command = new RelayCommand(_ =>
+                {
+                    msgbox.Result = i_copy;
+                    msgbox.Close();
+                }),
+            });
```

Comment:
1. The purpose of `i_copy` is unobvious for those who don't know about capture issues in C# (I didn't and had to guess and google for example).
2. Naming violation.

```suggestion
            // passing just i to the lambda will not work as
            // intended due to C# variable capture specifics
            int iCaptured = i;
            commands.Add(new CommandViewModel()
            {
                Text = selections[i],
                Command = new RelayCommand(_ =>
                {
                    msgbox.Result = iCaptured;
                    msgbox.Close();
                }),
            });
```